lib/dest/writeContents/index.js

```diff
@@ -44,12 +44,12 @@ function writeContents(writePath, file, cb) {
       if (err) {
         return complete(err);
       }
-      // octal 7777 = decimal 4095
-      var currentMode = (st.mode & 4095);
-      if (currentMode === file.stat.mode) {
+      var currentMode = (st.mode & parseInt('0777', 8));
+      var expectedMode = (file.stat.mode & parseInt('0777', 8));
+      if (currentMode === expectedMode) {
```

Comment:
The file is written before the chmod check, written() is the callback of the write* https://github.com/wearefractal/vinyl-fs/blob/ba57ce76e978850f79f78bba9b8aa1a73ac5fb2f/lib/dest/writeContents/index.js#L10-22